rust_src/src/fonts.rs

```diff
@@ -97,6 +103,70 @@ impl FontExtraType {
     }
 }
 
+pub type LispFontObjectRef = ExternalPtr<Lisp_Font_Object>;
+
+impl LispFontObjectRef {
+    pub fn as_lisp_obj(self) -> LispObject {
+        LispObject::tag_ptr(self, Lisp_Type::Lisp_Vectorlike)
+    }
+
+    pub fn add_log(self, action: &str, result: LispObject) {
+        let c_str = CString::new(action).unwrap();
+        unsafe { font_add_log(c_str.as_ptr(), self.as_lisp_obj(), result) }
+    }
+
+    pub fn close(mut self, mut frame: LispFrameRef) {
+        if data::aref(self.as_lisp_obj(), FONT_TYPE_INDEX.into()).is_nil() {
+            // Already closed
+            return;
+        }
+        self.add_log("close", LispObject::from(false));
+        unsafe {
+            if let Some(f) = (*self.driver).close {
+                f(self.as_mut())
+            }
+            let mut display_info = &mut *(*frame.output_data.x).display_info;
+            debug_assert!(display_info.n_fonts > 0);
+            display_info.n_fonts -= 1;
+        }
+    }
+}
+
+impl From<LispFontObjectRef> for LispObject {
+    fn from(f: LispFontObjectRef) -> Self {
+        f.as_lisp_obj()
+    }
+}
+
+impl From<LispObject> for LispFontObjectRef {
+    fn from(o: LispObject) -> Self {
```

Comment:
I think the `as_lisp_obj` idiom is tech debt and we should only use into/from instead.

Reply:
Thoughts?

Reply:
That's actually quite reasonable, it would be more slightly more idiomatic and safer in the long term to stick to from/into

Reply:
I think we already had this. I would say open an issue ?